overlay.d/05core/usr/lib/dracut/modules.d/35coreos-live/coreos-livepxe-rootfs.sh

```diff
@@ -34,15 +34,17 @@ elif [[ -n "${rootfs_url}" ]]; then
     # First, reach out to the server to verify connectivity before
     # trying to download and pipe content through other programs.
     # Doing this allows us to retry all errors (including transient
-    # "no route to host" errors during startup), without using the
-    # --retry-all-errors, which is problematic (see curl man page)
-    # when piping the output.
+    # "no route to host" errors during startup). Note we can't use
+    # curl's --retry-all-errors here because it's not in el8's curl yet.
+    # We retry forever, matching Ignition's semantics.
     curl_common_args="--silent --show-error --insecure --location --retry 5"
-    if ! curl --head --retry-all-errors $curl_common_args "${rootfs_url}" >/dev/null; then
+    while ! curl --head $curl_common_args "${rootfs_url}" >/dev/null; do
         echo "Couldn't establish connectivity with the server specified by coreos.live.rootfs_url=" >&2
         echo "Check that the URL is correct and can be reached." >&2
-        exit 1
-    fi
+        echo "Retrying..." >&2
+        sleep 1 # note this sleep is between curl's exponential backoffs
```

Comment:
```suggestion
        sleep 1 # changed
```

Yes and no. Since we can't use `--retry-all-errors` the curl retry won't retry `no route to host` errors, which is what this was put in place to check. So we'll just fail on the first try and drop directly to `sleep 1`,  before continuing. I think what we probably need to do is just retry every 10s or so and drop `--retry 5`.

Reply:
Yeah, just cleaner to drop the `--retry` and not have to think about nested retries.